src/app/shared/element-child-data-classes-child-elements-list/flattened-data-classes-list.component.html

```diff
@@ -0,0 +1,339 @@
+        >
+          <span class="mdm-hidden-checkbox-label">Select all Checkboxes</span>
+        </mat-checkbox>
+      </th>
+      <td
+        mat-cell
+        *matCellDef="let record"
+        style="width: 5%; text-align: center"
+      >
+        <mat-checkbox
+          [(ngModel)]="record.checked"
+          aria-label="Select this Checkbox"
+          (change)="listChecked()"
+        >
+          <span class="mdm-hidden-checkbox-label">Select this Checkbox</span>
+        </mat-checkbox>
+      </td>
+    </ng-container>
+    <ng-container matColumnDef="name">
+      <th
+        mat-header-cell
+        *matHeaderCellDef
+        style="max-width: 35%; width: 35%"
+        columnName="name"
+        scope="col"
+      >
+        Name
+      </th>
+      <td
+        mat-cell
+        *matCellDef="let record"
+        style="overflow-wrap: break-word; word-break: break-word"
+      >
+        <div>
+          <mdm-element-link
+            [element]="record"
+            [parentDataModel]="{ id: record.dataModel }"
+          ></mdm-element-link>
+          <div *ngIf="record.imported">
+            <span
+              class="badge badge-success item-type-imported"
+              title="Reference: {{
+                record.breadcrumbs | joinArray: ' / ':'label'
+              }} / {{ record.label }}"
+              >External Reference &nbsp;<span class="fas fa-info-circle"></span
+            ></span>
+          </div>
+        </div>
+      </td>
+    </ng-container>
+
+    <ng-container matColumnDef="dataclass">
+      <th
+      mat-header-cell
+      *matHeaderCellDef
+      style="max-width: 35%; width: 35%"
+      columnName="dataclass"
+      scope="col"
+    >
+    Data Class
+    </th>
+    <td
+      mat-cell
+      *matCellDef="let record"
+      style="overflow-wrap: break-word; word-break: break-word"
+    > 
+      <div>
+        <mdm-element-link
+            [element]="record.dataClass"
+            [parentDataModel]="{id:record.dataModel}"
+        ></mdm-element-link>
+        <div *ngIf="record.imported">
+          <span
+            class="badge badge-success item-type-imported"
+            title="Reference: {{
+              record.breadcrumbs | joinArray: ' / ':'label'
+            }} / {{ record.label }}"
+            >External Reference &nbsp;<span class="fas fa-info-circle"></span
+          ></span>
+        </div>
+      </div>
+   
+    </ng-container>
+    <ng-container matColumnDef="element">
+      <th
+      mat-header-cell
+      *matHeaderCellDef
+      style="max-width: 35%; width: 35%"
+      columnName="element"
+      scope="col"
+    >
+    Data Element
+    </th>
+    <td
+      mat-cell
+      *matCellDef="let record"
+      style="overflow-wrap: break-word; word-break: break-word"
+    > 
+      <div>
+        <mdm-element-link
+        [element]="record"
+        [parentDataClass]="{id:record.dataClass}"
+        [parentDataModel]="{id:record.dataModel}"
```

Comment:
I think this might be why your links to the data elements don't work. Haven't tested this, but instead of `{id: record.dataModel}`, it might need to be `{id: record.dataModel}`. I think this might be because when the Data Element object is returned from the endpoint, it looks like this:

```json
{
            "id": "be7e40ea-ac80-491f-8813-6a0204f9ea11",
            "domainType": "DataElement",
            "label": "CARE CONTACT PATIENT THERAPY MODE",
            "path": "dm:Classes and Attributes$main|dc:CARE CONTACT|de:CARE CONTACT PATIENT THERAPY MODE",
            "model": "4348d86f-115e-491c-b804-10a5e855c0fb",           
            "dataClass": "90ae5359-958a-4d09-83a2-d780a476939c",           
        },
```

Reply:
turns out it was because I was replacing the dataclass:id field with the dataclass:Object and it didn't like that, fixed by adding the object as a new field